lib/setupmanagers/hckclient.rb

```diff
@@ -72,19 +73,20 @@ def set_machine_ready
       raise ClientRunError, "Couldn't set #{@name} state to Ready"
     end
 
-    def run_post_start_commands
-      @project.engine.drivers&.each do |driver|
-        driver.post_start_commands&.each do |command|
-          return unless command.guest_run
-
-          @logger.info("Running command (#{command.desc}) on client #{@name}")
-          @tools.run_on_machine(@name, command.desc, command.guest_run)
-          next unless command.guest_reboot
+    def post_start_commands
+      @project.engine.drivers.flat_map(&:post_start_commands).compact.select(&:guest_run).to_a +
+        @setup_manager.clients_vm[@name].post_start_commands&.select(&:guest_run).to_a
```

Comment:
will be broken with `physhck`. Better to define `post_start_commands` in setupmanager and move `clients_vm` logic inside